raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -180,6 +353,251 @@ public void testVoteRequestTimeout() throws Exception {
         assertEquals(ElectionState.withElectedLeader(epoch, localId), quorumStateStore.readElectionState());
     }
 
+    @Test
+    public void testHandleValidVoteRequestAsFollower() throws Exception {
+        int leaderEpoch = 2;
+        int otherNodeId = 1;
+        Set<Integer> voters = Utils.mkSet(localId, otherNodeId);
+
+        MockQuorumStateStore electionStore = new MockQuorumStateStore() {
+            @Override
+            public ElectionState readElectionState() {
+                return ElectionState.withUnknownLeader(leaderEpoch);
+            }
+        };
+        KafkaRaftClient client = buildClient(voters, electionStore);
+        assertFalse(client.quorumState().followerStateOrThrow().hasVoted());
+
+        handleVoteRequest(leaderEpoch, otherNodeId, client, true);
+
+        assertTrue(client.quorumState().isFollower());
+        assertTrue(client.quorumState().followerStateOrThrow().hasVoted());
```

Comment:
In other test cases, we validate state like this using the `QuorumStateStore`. I'm curious if you think that is not sufficient. It would be nice if we didn't need to expose `quorumState`.

Reply:
Will have an attempt.